interview-questions.md

```diff
@@ -1,57 +1,83 @@
- #ASSESSMENT 2: Interview Practice Questions
+# ASSESSMENT 2: Interview Practice Questions
 
 Answer the following questions.
 
 First, without external resources. Challenge yourself to answer from memory as if you were in a job interview.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-1. What is the difference between a parameter and an argument?
+1. What is the difference between a parameter and an argument? 
 
-Your answer:
+Your answer: Parameter is a variable, argument is the data that you are trying to convey and pass. 
 
-Researched answer:
+
+Researched answer: In a Parameter variables are used to translate arguments into functions. An argument Function parameters are the names listed in the function's definition. Function arguments are the values passed to the function. Parameters are passed to the values of the arguments supplied.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
 
-Researched answer:
+Your answer: The built in method.map will take an individual data set, an array. It takes the same 3 arguements like .forEach. It was always return as a new array, of the same length as the original array. It will do something about each item in the item. A return is required when using a the .map method.
+
+Researched answer: The .map built in method takes in 3 parameters are arguements, in one individual data set. It takes the same 3 arguements are .forEach but its reutrn a new array. [value, index, array] Index and Array are optional components in the data set.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: .map built in method with return the array to the original length of the inputted array. The .filter function is looking to minimize/simply the array by using what logic is inputted into the parameter. The .map will take logic the same as .filter but will not show the same results as .filter.
+
+Researched answer: The .map and .filter are both built in methods that perform specificed logic to create a desired outcome/output. The difference between the two is that .map will return the original size of the parameter, and .filter will return the a new array with only displaying the results. .filter is looking to shorten the array.
```

Comment:
.map and .filter are higher-order functions that are a little different than built-in methods.